service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -15,10 +15,11 @@ import (
 	"github.com/giantswarm/cluster-operator/service/controller/key"
 )
 
-type clusterProfile string
+type clusterProfile int
 
 const (
-	xs clusterProfile = "XS"
+	unknown clusterProfile = iota
+	xs
```

Comment:
nice I learnt about `iota`
But is not `unkown` a bit awkward for the var name?

Reply:
Yes still not keen on unknown.

What about undefined? As we've chosen to only define XS so far.

Reply:
Removed unknown, again. Unset/undefined models the unknown cluster profile. PTAL